### PR TITLE
Improve api

### PR DESCRIPTION
**Pull request by pranav-bhatt**
Improved API calls for cloudevents-sdk-actix-web

### CODE REVIEW

**Comment by slinkydeveloper** [comment]
@pranav-bhatt I see the changes to reqwest module too, can you rebase your branch on latest master?

**Comment by pranav-bhatt** [comment]
> 
> 
> @pranav-bhatt I see the changes to reqwest module too, can you rebase your branch on latest master?

I rebased the branch, but there are still a few commits from the reqwest, [which apparently is like so by design](https://stackoverflow.com/questions/16306012/github-pull-request-showing-commits-that-are-already-in-target-branch). Do correct me if this case doesn't apply for this.

File: cloudevents-sdk-actix-web/src/server_response.rs [hunk 1]
```diff
@@ -75,6 +76,25 @@ pub async fn event_to_response(
         .map_err(actix_web::error::ErrorBadRequest)
 }
 
+/// Extention Trait for [`Event`]which acts as a wrapper for the function [`event_to_response()`]
+#[async_trait(?Send)]
+pub trait EventExt {
```
Comment (slinkydeveloper) [inline cloudevents-sdk-actix-web/src/server_response.rs:81]: I would prefer to extend `HttpResponseBuilder` directly, like:

```rust
HttpResponseBuilder::new(StatusCode::OK)
  .event(event)
```

File: cloudevents-sdk-actix-web/src/server_request.rs [hunk 2]
```diff
@@ -111,6 +112,25 @@ pub async fn request_to_event(
         .map_err(actix_web::error::ErrorBadRequest)
 }
 
+/// Extention Trait for [`HttpRequest`]which acts as a wrapper for the function [`request_to_event()`]
+#[async_trait(?Send)]
```
Comment (slinkydeveloper) [inline cloudevents-sdk-actix-web/src/server_request.rs:116]: Can you check both here and in `HttpResponseBuilderExt` if it breaks to remove the `?Send`? Let's add it only if we need it.

File: cloudevents-sdk-actix-web/src/server_response.rs [hunk 3]
```diff
@@ -75,6 +76,25 @@ pub async fn event_to_response(
         .map_err(actix_web::error::ErrorBadRequest)
 }
 
+/// Extention Trait for [`HttpResponseBuilder`]which acts as a wrapper for the function [`event_to_response()`]
```
Comment (slinkydeveloper) [inline cloudevents-sdk-actix-web/src/server_response.rs:79]: You forgot a space at [`HttpResponseBuilder`]which

File: cloudevents-sdk-actix-web/src/server_request.rs [hunk 4]
```diff
@@ -111,6 +112,25 @@ pub async fn request_to_event(
         .map_err(actix_web::error::ErrorBadRequest)
 }
 
+/// Extention Trait for [`HttpRequest`]which acts as a wrapper for the function [`request_to_event()`]
```
Comment (slinkydeveloper) [inline cloudevents-sdk-actix-web/src/server_request.rs:115]: Same as below, you forgot a space

File: cloudevents-sdk-actix-web/src/server_request.rs (hunk 2 above)
Comment (pranav-bhatt) [reply cloudevents-sdk-actix-web/src/server_request.rs:116]: Yeah it brakes unfortunately without ```?Send```.

**Comment by pranav-bhatt** [comment]
> 
> 
> Just some minor nits
> 
> Also, does the example works properly?

yup the example works!

**Comment by slinkydeveloper** [comment]
Can you convert the pr to ready to review?

**Comment by pranav-bhatt** [comment]
> 
> 
> Can you convert the pr to ready to review?

done!

**Pull request merged**